### DISCUSSION TITLE
How to put the giu Main Loop in a go routine

### DISCUSSION THREAD

**Discussion by mtp** [AllenDang/giu#107] [Q&A] [answered]
Hello,

I would like to have the UI running in its own thread, while the app controller is doing some work as the same time.
When I cann the Main Loop Running function via a go routine, it crash.
Did you already tried that ?

**Comment by AllenDang** [comment, marked as answer]
No, you cannot run main loop in another thread. You can run UI in main thread and run other logic in its own go routine. What's your use case?

**Reply by mtp** [reply]
I wanted to create a MVC pattern, but I guess the view have to be the main thread.

**Comment by ilius** [comment]
Would `sync.WaitGroup` help?
To exit only after all go routines (UI and MVC) are done.
https://golang.org/pkg/sync/#WaitGroup

**Reply by mtp** [reply]
I don't know. Actually I want the controller to run in the main thread, while the UI run in another thread.
